Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -118,15 +118,126 @@ public static bool IsBase64EncodingOfPrintableCharacters(string secret)
             return true;
         }
 
-        public static bool ContainsDigitAndChar(string matchedPattern)
+        /// <summary>
+        /// Determines whether the input string contains both digit and letter.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        /// <returns>true if the input string contains both digit and letter; otherwise, false.</returns>
+        public static bool ContainsDigitAndChar(string text)
+        {
+            // TODO, change the function name as ContainsDigitAndLetter and also fixed the 44 references
+            return ContainsMinimumCountOfDigits(text, 1) && ContainsMinimumCountOfLetters(text, 1);
+        }
+
+        /// <summary>
+        /// Determines whether the input string contains both lowercase and uppercase letter.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        /// <returns>true if the input string contains both lowercase and uppercase letter; otherwise, false.</returns>
+        public static bool ContainsLowercaseAndUppercaseLetter(string text)
+        {
+            return ContainsMinimumCountOfLowercaseLetters(text, 1) && ContainsMinimumCountOfUppercaseLetters(text, 1);
+        }
+
+        /// <summary>
+        /// Determines whether the input string contains mininum count of digits.
+        /// </summary>
+        /// <param name="text">The input string.</param>
+        /// <param name="lowerBound">The minimum count of digits required.</param>
+        /// <returns>true if the input string contains minimum count of digits; otherwise, false.</returns>
+        public static bool ContainsMinimumCountOfDigits(string text, int lowerBound)
```

Comment:
>ContainsMinimumCountOfDigits [](http://example.com/codeflow?start=27&length=28)

do we actually need this functionality?

Reply:
MICROSOFT AAD APPLICATION KEY needs at least 2 digits